api/mesh/v1alpha1/traffic_route.proto

```diff
@@ -137,15 +137,83 @@ message TrafficRoute {
       map<string, StringMatcher> headers = 3;
     }
 
+    // Modify defines modifications of matched HTTP messages
+    message Modify {
+
+      // RegexReplace defines a way to match string using regex and build a new
+      // one using subsitution section
+      message RegexReplace {
+        // Pattern of the regex using RE2 syntax
+        // https://github.com/google/re2/wiki/Syntax
+        string pattern = 1;
+        // Substitution using regex groups. E.g. use \\1 as a first matched
+        // group
+        string substitution = 2;
+      }
+
+      // Path defines modification of path of the HTTP request
+      message Path {
+        oneof type {
+          // rewritePrefix rewrites previously matched prefix in match section
+          string rewritePrefix = 1;
+          // regex rewrites prefix using regex with substitution
+          RegexReplace regex = 2;
+        }
+      }
+
+      // Host defines modification of the HTTP Host header
+      message Host {
+        oneof type {
+          // value replaces the host header with given value
+          string value = 1;
+          // fromPath replaces the host header from path using regex
+          RegexReplace fromPath = 2;
+        }
+      }
+
+      // Headers defines modification of HTTP headers
+      message Headers {
+        // Add defines operation of adding new HTTP header
+        message Add {
+          // Name of the header
+          string name = 1;
+          // Value of the header
+          string value = 2;
+          // If true, it appends the value if there is already a value
```

Comment:
Otherwise it replaces all existing values with this new one?

Reply:
Yes. Let me clarify this in the comment.

Reply:
also added e2e test